### PR TITLE
🐛 Python CDK: fix `StopIteration` error for `check_availability`

### PR DESCRIPTION
**Pull request by bazarnov**
## What

Fixed missing handling of `StopIteration` after: https://github.com/airbytehq/airbyte/pull/19977
____

## How
- added try/except statement to handle `StopIteration` error for empty streams

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/utils/stream_helpers.py [hunk 1]
```diff
@@ -17,10 +17,13 @@ def get_first_record(self, stream: Stream) -> StreamData:
...
-        # Some streams need a stream slice to read records (e.g. if they have a SubstreamSlicer)
-        stream_slice = self.get_stream_slice(stream)
-        records = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
-        next(records)
+        try:
+            # Some streams need a stream slice to read records (e.g. if they have a SubstreamSlicer)
+            stream_slice = self.get_stream_slice(stream)
+            records = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
+            return next(records)
+        except StopIteration:
```
Comment (grubberr) [inline airbyte-cdk/python/airbyte_cdk/sources/utils/stream_helpers.py:25]: maybe just use `next(records, None)` ?

**Review by erohmensing — APPROVED** [review]
Added a test and confirmed that the test fails on master but passes with your fix. LGTM, thank you for catching!

**Review by grubberr — APPROVED** [review]

File: airbyte-cdk/python/airbyte_cdk/sources/utils/stream_helpers.py (hunk 1 above)
Comment (grubberr) [reply airbyte-cdk/python/airbyte_cdk/sources/utils/stream_helpers.py:25]: I am boring man today `next(records, {})` ? :)

**Review by roman-yermilov-gl — APPROVED** [review]

**Pull request merged**